components/clarinet-format/src/formatter/mod.rs

```diff
@@ -358,35 +358,71 @@ impl<'a> Aggregator<'a> {
         result
     }
 
+    // (define-trait trait-name (
+    //   (func1-name
+    //     (arg1-type arg2-type ...)
+    //     (return-type)
+    //   )
+    //   (func2-name (arg1-type arg2-type ...) (return-type))
+    // )
     fn define_trait(&self, exprs: &[PreSymbolicExpression], previous_indentation: &str) -> String {
         let mut acc = "(define-trait ".to_string();
         let indentation = &self.settings.indentation.to_string();
         let space = format!("{}{}", indentation, previous_indentation);
         acc.push_str(&self.format_source_exprs(slice::from_ref(&exprs[1]), previous_indentation));
+        acc.push_str(" (");
         acc.push('\n');
         acc.push_str(&space);
-        acc.push('(');
         let mut iter = exprs[2].match_list().unwrap().iter().peekable();
         while let Some(expr) = iter.next() {
+            let trailing = get_trailing_comment(expr, &mut iter);
             if let Some(list) = expr.match_list() {
-                let trailing = get_trailing_comment(expr, &mut iter);
-                acc.push('\n');
-                acc.push_str(&space);
-                acc.push_str(indentation);
-                acc.push_str(&self.format_list(list, indentation));
+                acc.push('(');
+                // name
+                acc.push_str(&self.display_pse(&list[0], previous_indentation));
+                let double = format!("{}{}", space, indentation);
+                // Process each element in the list after the method name
+                let mut i = 1;
+                while i < list.len() {
+                    if let Some(element_list) = list[i].match_list() {
+                        // Found a list (either args or return type)
+                        acc.push('\n');
+                        acc.push_str(&double);
+                        acc.push_str(&self.format_list(element_list, &double));
+
+                        // Check if next item is a comment
+                        if i + 1 < list.len() && is_comment(&list[i + 1]) {
```

Comment:
Would it be worth refactoring this logic, to call a function in both scenarios?

Reply:
yeah, I think this whole index junk is a bad idea.. Let me try a `iter`

Reply:
Refactored to use peekable like everything else 👍